caffe2/python/pybind_state.cc

```diff
@@ -1219,13 +1219,15 @@ void addGlobalMethods(py::module& m) {
 
         // Parse protobuffers to NetDefs
         std::vector<std::unique_ptr<caffe2::NetDef>> nets;
```

Comment:
Where is this variable used again?

Reply:
It is not used. It's used as a container for the unique_ptr to the nets. If the InferTensorShape... function has signature that takes vector<unique_ptr> then you cannot call it with existing structures, only newly created. I changed it to taking just vector<Netdef*> .

Reply:
Looking at the code, `nets` is useless and you can just remove it.

Reply:
Then I'd need to manually delete the created Nets?

Reply:
OK, you cannot. You need that to hold ownership of the net.

Reply:
Yes that's what I meant.